### PR TITLE
Add new section "Classes" to the node page

### PR DESCRIPTION
**Pull request by juliengk**
This new section display all the classes from the catalog

### CODE REVIEW

**Comment by juliengk** [comment]
puppetboard_classes

**Comment by daenney** [comment]
Cool. The layout feels a bit nasty now though, maybe move reports to the right?

**Comment by juliengk** [comment]
facter list is small because there was a filter

**Comment by dklueh** [comment]
I tried this in our environment- disyplaying classes would be a great feature.
But it doesn't work, error log:

[Tue Mar 10 08:21:04.696872 2015] [:error] [pid 2413:tid 140224524949248] [client 10.18.2.25:63192] mod_wsgi (pid=2413): Target WSGI script '/var/www/puppetboard/wsgi.py' cannot be loaded as Python module., referer: http://....loc/nodes
[Tue Mar 10 08:21:04.696935 2015] [:error] [pid 2413:tid 140224524949248] [client 10.18.2.25:63192] mod_wsgi (pid=2413): Exception occurred processing WSGI script '/var/www/puppetboard/wsgi.py'., referer: http://....loc/nodes
[Tue Mar 10 08:21:04.696957 2015] [:error] [pid 2413:tid 140224524949248] [client 10.18.2.25:63192] Traceback (most recent call last):, referer: http://....loc/nodes
[Tue Mar 10 08:21:04.696977 2015] [:error] [pid 2413:tid 140224524949248] [client 10.18.2.25:63192]   File "/var/www/puppetboard/wsgi.py", line 6, in <module>, referer: http://....loc/nodes
[Tue Mar 10 08:21:04.697031 2015] [:error] [pid 2413:tid 140224524949248] [client 10.18.2.25:63192]     from puppetboard.app import app as application, referer: http://puppetmaster.continentale.loc/nodes
[Tue Mar 10 08:21:04.697043 2015] [:error] [pid 2413:tid 140224524949248] [client 10.18.2.25:63192]   File "/usr/local/lib/python2.7/dist-packages/puppetboard/app.py", line 23, in <module>, referer: http://....loc/nodes
[Tue Mar 10 08:21:04.697125 2015] [:error] [pid 2413:tid 140224524949248] [client 10.18.2.25:63192]     from puppetboard.utils import (, referer: http://....loc/nodes
[Tue Mar 10 08:21:04.697146 2015] [:error] [pid 2413:tid 140224524949248] [client 10.18.2.25:63192] ImportError: cannot import name limit_reports, referer: http://....loc/nodes

**Comment by daenney** [comment]
That sounds to me like you have a broken checkout of Puppetboard. It's complaining about an ImportError for `limit_reports` but it is imported at the top of that file: https://github.com/juliengk/puppetboard/blob/node_classes/puppetboard/app.py#L25

**Comment by dklueh** [comment]
Yes, In added this limit_reports, but the I'll get an internal server error with the log-entry above

**Comment by daenney** [comment]
The error above is an `ImportError`:

> [Tue Mar 10 08:21:04.697146 2015] [:error] [pid 2413:tid 140224524949248] [client 10.18.2.25:63192] ImportError: cannot import name limit_reports, referer: http://....loc/nodes

I have not seen this error on my systems, so I'm pretty sure this PR isn't the source of your troubles.

File: puppetboard/app.py [hunk 1]
```diff
@@ -176,13 +177,23 @@ def node(node_name):
     node. This includes facts and reports but not Resources as that is too
     heavy to do within a single request.
     """
+    def classes(name=None):
+        for resource in puppetdb.catalog(name).resources:
```
Comment (daenney) [inline puppetboard/app.py]: What happens when `name=None`? If that's never supposed to happen, it shouldn't be a `kwarg`.

Comment (juliengk) [inline puppetboard/app.py]: you are right. I will change it when i get to work. thanks

**Comment by dklueh** [comment]
I added the diff with name=none, but the error might be the same:
mod_wsgi (pid=9797): Exception occurred processing WSGI script '/var/www/puppetboard/wsgi.py'., referer: http://.loc/
 Traceback (most recent call last):, referer: http://.loc/
 File "/var/www/puppetboard/wsgi.py", line 6, in <module>, referer: http://.loc/
   from puppetboard.app import app as application, referer: http://.loc/
   File "/usr/local/lib/python2.7/dist-packages/puppetboard/app.py", line 23, in <module>, referer: http://.loc/
     from puppetboard.utils import (, referer: http://.loc/
 ImportError: cannot import name limit_reports, referer: http://.loc/
 mod_wsgi (pid=9798): Target WSGI script '/var/www/puppetboard/wsgi.py' cannot be loaded as Python module.
mod_wsgi (pid=9798): Exception occurred processing WSGI script '/var/www/puppetboard/wsgi.py'.
 Traceback (most recent call last):
File "/var/www/puppetboard/wsgi.py", line 6, in <module>
from puppetboard.app import app as application
File "/usr/local/lib/python2.7/dist-packages/puppetboard/app.py", line 23, in <module>
from puppetboard.utils import (
ImportError: cannot import name limit_reports

**Comment by daenney** [comment]
Look at the error:

```
File "/usr/local/lib/python2.7/dist-packages/puppetboard/app.py", line 23, in 
from puppetboard.utils import (
ImportError: cannot import name limit_reports
```

Do you have a `def limit_reports` in `puppetboard/utils.py`?

**Comment by dklueh** [comment]
No, there is no definition like that.
Should it be added?

**Comment by daenney** [comment]
Yes and no. You've applied this patch, which was raised against the `master` branch on a Puppetboard version that is not up to date with current `master`. This is a problem on your side, not on the side of this pull request. Please update your Puppetboard version first before filing any further issues.

I'm going to remove the rest of the comments as they create unnecessary noise for something that has nothing to do with this PR.

**Pull request closed without merging**